balancer-js/src/modules/pools/pool-types/concerns/composableStable/join.concern.integration.spec.ts

```diff
@@ -1,209 +1,165 @@
-      const transactionResponse = await sendJoinTransaction(
-        ...attributesValues,
-        { value }
-      );
-      transactionReceipt = await transactionResponse.wait();
+  it('should return correct attributes for joining', async () => {
+    const tokensIn = removeItem(pool.tokensList, pool.bptIndex);
+    const amountsIn = tokensIn.map((_, i) =>
+      parseFixed((i * 100).toString(), 18).toString()
+    );
+    const slippage = BigNumber.from('6');
+    const { attributes, functionName } = pool.buildJoin(
+      signerAddress,
+      tokensIn,
+      amountsIn,
+      slippage.toString()
+    );
 
-      bptMinBalanceIncrease = BigNumber.from(minBPTOut);
-      [bptBalanceAfter, ...tokensBalanceAfter] = await getBalances(
-        [pool.address, ...tokensWithoutBpt],
-        signer,
-        signerAddress
-      );
-    });
-
-    it('should work', async () => {
-      expect(transactionReceipt.status).to.eql(1);
-    });
-
-    it('should increase BPT balance', async () => {
-      expect(bptBalanceAfter.sub(bptBalanceBefore).gte(bptMinBalanceIncrease))
-        .to.be.true;
-    });
-
-    it('should decrease tokens balance', async () => {
-      for (let i = 0; i < tokensWithoutBpt.length; i++) {
-        expect(
-          tokensBalanceBefore[i].sub(tokensBalanceAfter[i]).toString()
-        ).to.equal(amountsIn[i]);
-      }
-    });
+    expect(functionName).to.eq('joinPool');
+    expect(attributes.poolId).to.eq(testPoolId);
+    expect(attributes.recipient).to.eq(signerAddress);
+    expect(attributes.sender).to.eq(signerAddress);
+    expect(attributes.joinPoolRequest.assets).to.deep.eq(pool.tokensList);
+    expect(attributes.joinPoolRequest.fromInternalBalance).to.be.false;
+    expect(attributes.joinPoolRequest.maxAmountsIn).to.deep.eq(
+      insert(amountsIn, pool.bptIndex, '0')
+    );
   });
 
-  context('join transaction - single token join', () => {
-    const amountsIn = [parseFixed('0.001', 18).toString()];
-    const slippage = '10';
-    it('should fail on number of input tokens', async () => {
-      let errorMessage;
-      try {
-        pool.buildJoin(
-          signerAddress,
-          tokensWithoutBpt.map((t) => t),
-          amountsIn,
-          slippage
-        );
-      } catch (error) {
-        errorMessage = (error as Error).message;
-      }
-      expect(errorMessage).to.contain(
-        BalancerError.getMessage(BalancerErrorCode.INPUT_LENGTH_MISMATCH)
-      );
-    });
+  it('should automatically sort tokens/amounts in correct order', async () => {
+    const tokensIn = removeItem(pool.tokensList, pool.bptIndex);
+    const amountsIn = tokensIn.map((_, i) =>
+      parseFixed((i * 100).toString(), 18).toString()
+    );
+    const slippage = BigNumber.from('6');
+    const attributesSorted = pool.buildJoin(
+      signerAddress,
+      tokensIn,
+      amountsIn,
+      slippage.toString()
+    );
+    const attributesUnSorted = pool.buildJoin(
```

Comment:
I think this makes more sense to be "parametersReversed", and the other one just "parameters":
- "parameters" instead of "attributes" because they are transaction parameters, and it could get confused with the JoinPool Attributes (the one with sender, recipient, joinPoolRequest..);
- "Reversed" instead of "Sorted" and "Unsorted" because when you reverse you are doing a kind of a sorting, for me the Unsorted should be the one that you don't reverse the tokensIn and amountsIn, it could get a little confuse using these words, I think.

Reply:
Yeah I see what you mean. Was tricky to find a good solution so I've renamed them attributesA/B and added comments to explain.